cmd/image-builder/main.go

```diff
@@ -356,6 +362,27 @@ func buildInADO(o options) error {
 	return nil
 }
 
+// validateDockerFile checks whether provided dockerfile
+// will run smoothly in ADO image-builder.
+// It return clear error message, including known issue urls.
+func validateDockerFile(dockerFilePath string) error {
+	// Load Dockerfile
+	data, err := os.ReadFile(dockerFilePath)
+	if err != nil {
+		return fmt.Errorf("cannot load provided dockerfile: %w", err)
+	}
+
+	// Dockerfile using /workspace
+	// see: https://github.com/GoogleContainerTools/kaniko/issues/2192
+	// see: https://github.com/GoogleContainerTools/kaniko/issues/3176
+	re := regexp.MustCompile("WORKDIR.*/workspace")
```

Comment:
As far I remember the docker file which uses a destination path starting with "/workspace" for COPY command will be impacted by this issue too. Can we try to detect such case as well?

Reply:
In theory, they shouldn't copy from other directory than workdir in previous step, but as precaution added COPY command as well.